docs/architecture.md

```diff
@@ -0,0 +1,25 @@
+# Architecture
+
+## The chosen architecture
+libcal-proxy is a [vert.x](https://vertx.io/) application.
+It consists of:
+* A verticle that 
+** Loads application configuration
+** Retrieves initial OAuth tokens
+** Deploys the two request handlers
+* A status handler, called to determine if the application is running
+* A LibCal Proxy handler, which
+** Receives LibCal API requests
+** Retrieves an OAuth token from application storage
+** Passes the API request along to LibCal with the token
+** Returns the LibCal response to the original client
```

Comment:
I feel like this part should also mention the OAuthTokenService (which is what retrieves the OAuth tokens, not MainVerticle) and LibCalProxyService

Reply:
MainVerticle deploys those two services

Reply:
Added info on services, tweaked descriptions of verticle/handler to reference services.